encfs/CipherFileIO.cpp

```diff
@@ -546,7 +546,7 @@ ssize_t CipherFileIO::read(const IORequest &origReq) const {
     memcpy(req.data, &headerBuf[headerOffset], headerBytes);
 
     // the read does not want data beyond the header
-    if (headerBytes == req.dataLen) {
+    if ((size_t)headerBytes == req.dataLen) {
```

Comment:
Ok I looked through all of the casts again. Everything seems safe except this one. It looks like headerBytes may be negative at this point. Rather cast req.dataLen to int?

Reply:
Mmmmh I'm not sure how `headerBytes` could be negative.
Which path would make it negative ?

I think casting `size_t` to `int` is rather dangerous.

Thx 👍